go/build.sh

```diff
@@ -1,131 +1,110 @@
+	socialapi/workers/cmd/realtime/gatekeeper
+	socialapi/workers/cmd/realtime/dispatcher
+	socialapi/workers/cmd/topicfeed
+	socialapi/workers/cmd/migrator
+	socialapi/workers/cmd/sitemap/sitemapfeeder
+	socialapi/workers/cmd/sitemap/sitemapgenerator
+	socialapi/workers/cmd/sitemap/sitemapinitializer
+	socialapi/workers/cmd/algoliaconnector
+	socialapi/workers/cmd/algoliaconnector/deletedaccountremover
+	socialapi/workers/payment/paymentwebhook
+	socialapi/workers/cmd/topicmoderation
+	socialapi/workers/cmd/collaboration
+	socialapi/workers/cmd/email/activityemail
+	socialapi/workers/cmd/email/dailyemail
+	socialapi/workers/cmd/email/privatemessageemailfeeder
+	socialapi/workers/cmd/email/privatemessageemailsender
+	socialapi/workers/cmd/email/emailsender
+	socialapi/workers/cmd/team
+	socialapi/workers/cmd/integration/webhook
+	socialapi/workers/algoliaconnector/tagmigrator
+	socialapi/workers/algoliaconnector/contentmigrator
+	socialapi/workers/cmd/integration/eventsender
+	socialapi/workers/cmd/integration/webhookmiddleware
+
+	vendor/github.com/koding/kite/kitectl
+	vendor/github.com/canthefason/go-watcher
+	vendor/github.com/mattes/migrate
+	vendor/github.com/alecthomas/gocyclo
+	vendor/github.com/remyoudompheng/go-misc/deadcode
+	vendor/github.com/jteeuwen/go-bindata/go-bindata
 )
 
-
-`which go` install -v -ldflags "$ldflags" "${services[@]}"
-
-cd $GOPATH
-mkdir -p build/broker
-cp bin/broker build/broker/broker
-
-# build terraform services
-terraformservices=(
-  koding/kites/cmd/provider-vagrant
-
-  vendor/github.com/hashicorp/terraform/builtin/bins/provider-aws
-  vendor/github.com/hashicorp/terraform/builtin/bins/provider-terraform
-  vendor/github.com/hashicorp/terraform/builtin/bins/provider-null
-  vendor/github.com/koding/terraform-provider-github/cmd/provider-github
-
-  vendor/github.com/hashicorp/terraform/builtin/bins/provisioner-file
-  vendor/github.com/hashicorp/terraform/builtin/bins/provisioner-local-exec
-  vendor/github.com/hashicorp/terraform/builtin/bins/provisioner-remote-exec
+export TERRAFORM_COMMANDS=(
+	vendor/github.com/hashicorp/terraform/builtin/bins/provider-aws
+	vendor/github.com/hashicorp/terraform/builtin/bins/provider-terraform
+	vendor/github.com/hashicorp/terraform/builtin/bins/provider-null
+	vendor/github.com/hashicorp/terraform/builtin/bins/provisioner-file
+	vendor/github.com/hashicorp/terraform/builtin/bins/provisioner-local-exec
+	vendor/github.com/hashicorp/terraform/builtin/bins/provisioner-remote-exec
+	koding/kites/cmd/provider-vagrant
+	vendor/github.com/koding/terraform-provider-github/cmd/provider-github
 )
 
-tldflags="-X main.GitCommit${LINK_OPERATOR}${version:0:8}"
-`which go` install -v -ldflags "$tldflags"  "${terraformservices[@]}"
+# source configuration for kloud providers
+for provider in $KODING_REPO/go/src/koding/kites/kloud/provider/*; do
+	if [[ -d "${provider}/build.sh.d" ]]; then
+		for src in ${provider}/build.sh.d/*; do
+			if [[ -f "$src" ]]; then
+				source "$src"
```

Comment:
in general cases a `if [ -r "$src" ]` check would be a good practice but i think we can omit it here since we will manage the build scripts.